table/tables/index.go

```diff
@@ -103,236 +103,299 @@ func (c *index) GenIndexValue(sc *stmtctx.StatementContext, distinct bool, index
 	return tablecodec.GenIndexValuePortal(sc, c.tblInfo, c.idxInfo, c.needRestoredData, distinct, false, indexedValues, h, c.phyTblID, restoredData)
 }
 
+func (c *index) getIndexedValue(indexedValues []types.Datum) [][]types.Datum {
```

Comment:
If the input values are `[1, '[1, 2]']`, then the output should be `[[1, 1], [1, 2]]`?

Reply:
yes!

Reply:
Could you add some comments for the format of the output?